database-commons/src/main/java/io/cdap/plugin/db/ConnectionConfig.java

```diff
@@ -111,17 +132,55 @@ public Properties getConnectionArguments() {
     return arguments;
   }
 
+  /**
+   * Returns all configuration properties including database-specific ones as single string.
+   * In the case when there are no connection arguments, an empty string will be returned.
+   * @return connection arguments as string with '=' as key-value delimiter and ';' as pair delimiter.
+   */
+  public String getConnectionArgumentsString() {
+    if (this.connectionArguments == null) {
+      return getDBSpecificArgumentsString();
+    }
+
+    if (getDBSpecificArguments().isEmpty()) {
+      return this.connectionArguments;
+    }
+
+    return this.connectionArguments + ";" + getDBSpecificArgumentsString();
+  }
+
   /**
    * Constructs a connection string from host, port and database properties in a database-specific format.
    * @return connection string specific to a particular database.
    */
   public abstract String getConnectionString();
 
+  /**
+   * Returns list of initialization queries as a single string separated by ';' character. Initialization queries
+   * supposed to be executed preserving order right after connection establishing.
+   * In the case when there are no initialization queries, an empty string will be returned.
+   * @return list of initialization queries as a single string separated by ';' character.
+   */
+  public String getInitQueriesString() {
```

Comment:
similar comment here. It's less error prone if this returns a List rather than an encoded string.